Feat/adding coins address automatically

doha-pro opened the pull request:
adding the fature of adding the coin token address automatically 

- when the user is not connected to the wallet and chooses a coin for swapping a copy icon button appears to copy the token address 
- when the user is connected to the wallet a metamask icon appears next to the chosen coin and when pressed it adds the coin token address automatically

erwin-huang commented on src/components/CurrencyInputPanel/index.tsx:
```diff
@@ -1,9 +1,11 @@
 import React, { useState, useCallback } from 'react'
-import { Currency, Pair } from '@koda-finance/summitswap-sdk'
+import { Currency, Pair, Token  } from '@koda-finance/summitswap-sdk'
```
We don't need to update this component,
We just need to update `src/pages/Swap/index.tsx`

if `currencies[Field.OUTPUT]` is changed,
we can `registerToken` to metamask automatically.

if the user has the token registered, then we don't need to register the token anymore.

erwin-huang commented on src/components/SearchModal/CurrencyList.tsx:
```diff
@@ -2,7 +2,10 @@ import { Currency, CurrencyAmount, currencyEquals, ETHER, Token } from '@koda-fi
 import { VariableSizeList } from 'react-window'
 import styled from 'styled-components'
-import { Flex, Text } from '@koda-finance/summitswap-uikit'
+import { Flex, Text  } from '@koda-finance/summitswap-uikit'
```
We don't need to update this component too,
We just need to update `src/pages/Swap/index.tsx`

if `currencies[Field.OUTPUT]` is changed,
we can `registerToken` to metamask automatically.

if the user has the token registered, then we don't need to register the token anymore.

erwin-huang commented on src/components/CopyButton/index.tsx:
```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
```
We don't need `CopyButton` for this task

erwin-huang commented on src/connectors/index.ts:
```diff
@@ -81,3 +81,22 @@ export const walletlink = new WalletLinkConnector({
+
+
+
+export const registerToken = async (tokenAddress: string, tokenSymbol: string, tokenDecimals: number) => {
```
I think it's better to pass `Token` to this method
`export const registerToken = async (token: Token)`

doha-pro replied:
it worked well ... resolved

erwin-huang commented on src/components/Svg/Icons/Metamask.tsx:
```diff
@@ -0,0 +1,74 @@
+import React from "react";
```
We don't need the `Svg` folder. the theme, util, and icons files are unnecessary,  We use `src/img` for all the images

doha-pro replied:
I haven't used an image for the Metamask icon I used svg of metamask as it would be better in the ui and there wasn't one provided in the uikit that's why I made the svg

erwin-huang replied:
You can put the metamask svg on on `src/img`, and the other styles on `Svg` files,  we didn't use any of them right?

doha-pro replied:
I will do so and make sure that i don't need the other files and delete the un-necessary files

doha-pro replied:
I moved the `svg` folder to `src//img` and deleted the unused files they weren't a lot and left the files that this metamask svg depends on and not in the uikit

erwin-huang commented:
The conflict need to be resolved.

doha-pro commented on src/utils/getLpAddress.ts:
```diff
@@ -0,0 +1,27 @@
+import { Token, Pair, ChainId } from '@koda-finance/summitswap-sdk'
```
Ah it was a mistake will trying to know  the app

doha-pro commented on src/utils/bigNumber.ts:
```diff
@@ -0,0 +1,12 @@
+import BigNumber from 'bignumber.js'
```
Ah it was a mistake will trying to know  the app

doha-pro commented on src/state/user/hooks.tsx:
```diff
@@ -23,7 +23,7 @@ import {
 import { setThemeCache } from '../../utils/theme'
 
-function serializeToken(token: Token): SerializedToken {
+export function serializeToken(token: Token): SerializedToken {
```
i used it once will trying to achieve the task so i needed to be exported then i discovered that i don't need it and i forgot to remove it .. .resolved

doha-pro commented on src/react-app-env.d.ts:
```diff
@@ -34,3 +34,16 @@ interface BinanceChain {
   removeListener?: (method: string, listener: (...args: any[]) => void) => void
 }
+
+interface Window {
```
resolved

erwin-huang commented:
<img width="470" alt="Screen Shot 2022-03-25 at 14 33 59" src="https://user-images.githubusercontent.com/48702437/160075489-907bc647-fdf1-48d5-ae32-6face3190976.png">
Should align center

erwin-huang commented:
It's better to put the metamask icon on `summitswap-uikit`
You can see the sample here
https://github.com/Koda-Finance/summitswap-uikit/tree/main/src/components/Svg/Icons

doha-pro commented:
> <img alt="Screen Shot 2022-03-25 at 14 33 59" width="470" src="https://user-images.githubusercontent.com/48702437/160075489-907bc647-fdf1-48d5-ae32-6face3190976.png">
> 
> Should align center
![Screenshot from 2022-03-26 19-55-03](https://user-images.githubusercontent.com/96725421/160251656-25f13d09-3449-48c6-92aa-fa6c58a326cf.png)
![Screenshot from 2022-03-26 19-16-28](https://user-images.githubusercontent.com/96725421/160251663-7f7f34f5-2aa5-4bd8-aa77-21a0f6e954b7.png)


They are already aligned Center as in the picture

erwin-huang commented:
The icon  should be aligned center with the `CurrencySelect`.
<img width="453" alt="Screen Shot 2022-03-28 at 11 28 34" src="https://user-images.githubusercontent.com/48702437/160326843-619258bb-a724-4848-ac1c-9867b9f51ac7.png">

erwin-huang commented:
The conflict need to be resolved

doha-pro commented:
> The icon should be aligned center with the `CurrencySelect`. <img alt="Screen Shot 2022-03-28 at 11 28 34" width="453" src="https://user-images.githubusercontent.com/48702437/160326843-619258bb-a724-4848-ac1c-9867b9f51ac7.png">

I had fixed the alignment issue but in two ways ...
the with the buttons outside the `max` button interfere with it and ruins its alignment so i added a condition to the styling 
(That's why it's the same code for the 2 components but always the alignment differs)

which one is better to push its code 
![Screenshot from 2022-03-29 01-21-20](https://user-images.githubusercontent.com/96725421/160504023-1441fdc6-c014-44b8-866e-7e8f777806ec.png)
![Screenshot from 2022-03-28 23-54-39](https://user-images.githubusercontent.com/96725421/160504031-eeb8b2e9-5db6-422c-a36f-1e0af5cec32a.png)

doha-pro commented:
> The conflict need to be resolved

Yah ... Sorry haven't noticed because I have used the branch on the uikit after adding the  `Metamask icon`  .. 
I will resolve it with the final push 
Will it work even if the branch in the uikit (the one that contains the Metamask Icon) isn't merged yet ?

erwin-huang commented:
> > The conflict need to be resolved
> 
> Yah ... Sorry haven't noticed because I have used the branch on the uikit after adding the `Metamask icon` .. I will resolve it with the final push Will it work even if the branch in the uikit (the one that contains the Metamask Icon) isn't merged yet ?

You can resolve the conflict by choosing the version that you are currently working on.

We will merge the `uikit` and publish it after the interface is working great,
and we will update the interface's `uikit` version again after we publish the `uikit`

erwin-huang commented:
this one looks good 
<img width="798" alt="image" src="https://user-images.githubusercontent.com/48702437/160550483-34ced92c-97d8-45b5-b00b-f809af5a560e.png">
